src/Microsoft.Restier.WebApi/Query/QueryTotalCount.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Restier.Core.Query;
+
+namespace Microsoft.Restier.WebApi.Query
+{
+    internal class QueryTotalCount : IQueryExecutor
+    {
```

Comment:
My own opinion is that if this has nothing to do with OData/WebApi, it'll be much easier to reuse under some other circumstances. So I intentionally isolated any OData/WebApi specific logic here.
But I'd like to follow your design if you want it to be a WebApi dedicated implementation.